File: app/src/main/java/com/levnovikov/postbus/root/RootInteractor.java
package com.levnovikov.postbus.root;

import com.levnovikov.core_profile.UserRepository;
import com.levnovikov.postbus.root.di.RootScope;

import javax.inject.Inject;

/**
 * Author: lev.novikov
 * Date: 20/11/17.
 */

@RootScope
public class RootInteractor {

    private final UserRepository userRepository;
    private final RootRouter router;

    @Inject
    RootInteractor(
            RootRouter router,
            UserRepository userRepository) {
        this.userRepository = userRepository;
        this.router = router;
        onGetActive();
    }

    public void onGetActive() {
        checkUserData();
    }

    private void checkUserData() {
        userRepository.isUserLoggedIn()
                .subscribe(isLoggedIn -> {
                    if (isLoggedIn) {
                        loadProfile();
                    }
                }, this::handleInitError);
    }

    private void loadProfile() {
        userRepository.updateUserProfile()
                .subscribe(userProfile -> router.home(),
                        this::handleInitError);
    }

    private void handleInitError(Throwable error) {
        // TODO handle error
    }
}

File: app/src/main/java/com/levnovikov/postbus/root/RootInteractor.kt
package com.levnovikov.postbus.root

import com.levnovikov.core_profile.UserRepository
import com.levnovikov.postbus.root.di.RootScope
import javax.inject.Inject

/**
 * Author: lev.novikov
 * Date: 20/11/17.
 */

@RootScope
class RootInteractor @Inject constructor(
        private val router: RootRouter,
        private val userRepository: UserRepository) {

    init {
        onGetActive()
    }

    fun onGetActive() {
        checkUserData()
    }

    private fun checkUserData() {
        userRepository.isUserLoggedIn
                .subscribe({ isLoggedIn ->
                    if (isLoggedIn) {
                        loadProfile()
                    }
                }, { this.handleInitError(it) })
    }

    private fun loadProfile() {
        userRepository.updateUserProfile()
                .subscribe({ router.home() }, { this.handleInitError(it) })
    }

    private fun handleInitError(error: Throwable) {
        // TODO handle error
    }
}
